tests/functional/s3/test_ls_command.py

```diff
@@ -199,6 +199,40 @@ def test_requester_pays_with_no_args(self):
             'Prefix': 'foo/'
         })
 
+    def test_recursive_list_at_prefix(self):
```

Comment:
It may be worth adding a test for the in-between delimiter cases. For example:
```
$ aws s3 ls s3://mybucket/f --recursive
2017-11-22 12:48:50       1448 fo
2017-11-22 12:48:50       1448 foo/bar
```